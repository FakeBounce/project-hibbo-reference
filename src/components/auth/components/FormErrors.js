import React from 'react';
import { Dimensions } from 'react-native';
import { shape, string, number, bool } from 'prop-types';
import Error from './Error';

const { width } = Dimensions.get('window');

const FormErrors = ({ error, errorMsg, inputPosition, errorApi }) => {
  if (error) {
    return (
      <Error
        label={errorMsg}
        posY={inputPosition.y}
        posX={inputPosition.x + width * 0.08}
      />
    );
  } else if (errorApi !== '') {
    return (
      <Error
        label={errorApi}
        posY={inputPosition.y}
        posX={inputPosition.x + width * 0.08}
      />
    );
  }

  return null;
};

FormErrors.propTypes = {
  error: bool.isRequired,
  errorMsg: string.isRequired,
  inputPosition: shape({
    x: number.isRequired,
    y: number.isRequired,
  }).isRequired,
  errorApi: string.isRequired,
};

export default FormErrors;
