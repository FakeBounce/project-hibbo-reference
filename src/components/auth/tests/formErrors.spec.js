import React from 'react';
import renderer from 'react-test-renderer';

import FormErrors from '../components/FormErrors';

describe('FormErrors', () => {
  const position = {
    x: 0,
    y: 0,
  };

  it('should render the component', () => {
    const component = renderer
      .create(<FormErrors error={false} errorMsg="" errorApi="" inputPosition={position} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should render the component with API error', () => {
    const component = renderer
      .create(<FormErrors error={false} errorMsg="" errorApi="Error" inputPosition={position} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should render the component with an error', () => {
    const component = renderer
      .create(<FormErrors error errorMsg="Error" errorApi="" inputPosition={position} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
