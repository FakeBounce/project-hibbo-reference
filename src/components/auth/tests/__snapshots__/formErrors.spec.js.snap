// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrors should render the component 1`] = `null`;

exports[`FormErrors should render the component with API error 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "left": 50,
        "top": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "#FF7171",
        },
      ]
    }
  >
    <View
      style={undefined}
    >
      <View
        style={undefined}
      >
        <Image
          source={1}
          style={undefined}
        />
      </View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          Error
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          undefined,
          Object {
            "borderBottomColor": "#FF7171",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`FormErrors should render the component with an error 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "left": 50,
        "top": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "#FF7171",
        },
      ]
    }
  >
    <View
      style={undefined}
    >
      <View
        style={undefined}
      >
        <Image
          source={1}
          style={undefined}
        />
      </View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          Error
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          undefined,
          Object {
            "borderBottomColor": "#FF7171",
          },
        ]
      }
    />
  </View>
</View>
`;
